cpp/CMakeLists.txt

```diff
@@ -98,6 +98,10 @@ if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
   set(ARROW_CXXFLAGS "" CACHE STRING
     "Compiler flags to append when compiling Arrow")
 
+  option(ARROW_OPTIONAL_INSTALL
+    "If enabled install ONLY targets that have already been built"
```

Comment:
Can you add a comment that this is a kind of dangerous option? In the normal case, I would expect users to have turned features off completely via CMake flags; this is only something I would expect a core developer to use.

Reply:
Done